src/ioPlacer/src/DBWrapper.cpp

```diff
@@ -70,62 +73,42 @@ void DBWrapper::parseDEF(const std::string& filename)
   for (itr = libs.begin(); itr != libs.end(); ++itr) {
     searchLibs.push_back(*itr);
   }
-
-  _chip = defReader.createChip(searchLibs, filename.c_str());
 }
 
 void DBWrapper::populateIOPlacer()
 {
   _db = odb::dbDatabase::getDatabase(_parms->getDbId());
-  _chip = _db->getChip();
+  _tech = _db->getTech();
+  _block = _db->getChip()->getBlock();
   initNetlist();
   initCore();
 }
 
 void DBWrapper::initCore()
 {
-  odb::dbTech* tech = _db->getTech();
-  if (!tech) {
-    std::cout << "[ERROR] odb::dbTech not initialized! Exiting...\n";
-    std::exit(1);
-  }
-
-  int databaseUnit = tech->getLefUnits();
-
-  odb::dbBlock* block = _chip->getBlock();
-  if (!block) {
-    std::cout << "[ERROR] odb::dbBlock not found! Exiting...\n";
-    std::exit(1);
-  }
+  int databaseUnit = _tech->getLefUnits();
 
   odb::Rect rect;
-  block->getDieArea(rect);
+  _block->getDieArea(rect);
 
   Coordinate lowerBound(rect.xMin(), rect.yMin());
   Coordinate upperBound(rect.xMax(), rect.yMax());
 
   int horLayerIdx = _parms->getHorizontalMetalLayer();
   int verLayerIdx = _parms->getVerticalMetalLayer();
 
-  odb::dbTechLayer* horLayer = tech->findRoutingLayer(horLayerIdx);
+  odb::dbTechLayer* horLayer = _tech->findRoutingLayer(horLayerIdx);
   if (!horLayer) {
```

Comment:
dbTechLayers* is a pointer, not a boolean, so should use != nullptr, but that is just me being picky.